d2/src/main/java/com/linkedin/d2/discovery/stores/file/FileStore.java

```diff
@@ -185,7 +185,7 @@ public Map<String, T> getAll()
     List<String> propertyNames;
     try
     {
-      propertyNames = FileSystemDirectory.getFileListWithoutExtension(_fsPath);
+      propertyNames = FileSystemDirectory.getFileListWithoutExtension(_fsPath, _fsFileExtension);
```

Comment:
No need to change this any more with your recent backward compatible fix?

Reply:
This is still needed so that calls to this function in gRPC will use the correct file extension.

Reply:
got it, you are right.